loopout_test.go

```diff
@@ -376,3 +396,190 @@ func TestCustomSweepConfTarget(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestPreimagePush tests or logic that decides whether to push our preimage to
+// the server. First, we test the case where we have not yet disclosed our
+// preimage with a sweep, so we do not want to push our preimage yet. Next, we
+// broadcast a sweep attempt and push our preimage to the server. In this stage
+// we mock a server failure by not sending a settle update for our payment.
+// Finally, we make a last sweep attempt, push the preimage (because we have
+// not detected our settle) and settle the off chain htlc, indicating that the
+// server successfully settled using the preimage push. In this test, we need
+// to start with a fee rate that will be too high, then progress to an
+// acceptable one. We do this by starting with a high confirmation target with
+// a high fee, and setting the default confirmation fee (which our swap will
+// drop down to if it is not confirming in time) to a lower fee. This is not
+// intuitive (lower confs having lower fees), but it allows up to mock fee
+// changes.
+func TestPreimagePush(t *testing.T) {
```

Comment:
Isn't this just a `Success` test, as preimage push is part of the happy flow?

Reply:
In `TestSuccess` we push preimage once and then our payment is settled. This tests the case where we push and the payment is not settled, so we push again. It also fails to sweep on first try becuase fees are too high, whereas our success tests always sweep first thing. I can cut it, but I think it adds a few nice coverage improvements.